rhoc/pkg/cmd/deployments/validateSnapshot/validateSnapshot.go

```diff
@@ -0,0 +1,217 @@
+	}
+
+	cmdutil.AddClusterAmount(cmd, &opts.clusterAmount)
+	cmdutil.AddInputFile(cmd, &opts.inputFile)
+
+	return cmd
+}
+
+func run(opts *options) error {
+	fmt.Fprintln(opts.f.IOStreams.Out, "Starting snapshot validation")
+
+	c, err := service.NewAdminClient(&service.Config{
+		F: opts.f,
+	})
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintln(opts.f.IOStreams.Out, "Querying for clusters..")
+
+	clusterListOpts := request.ListOptions{
+		Page:     0,
+		Limit:    opts.clusterAmount,
+		AllPages: false,
+		OrderBy:  "created_at",
+		Search:   "state=ready",
+	}
+	clusters, err := service.ListClusters(c, clusterListOpts)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(opts.f.IOStreams.Out, "Got %d clusters\n", clusters.Size)
+
+	var snapshots []snapshot.DeploymentSnapshot
+	for _, cluster := range clusters.Items {
+		fmt.Fprintf(opts.f.IOStreams.Out, "Querying for deployments in cluster %s..\n", cluster.Id)
+
+		deployListOpts := request.ListDeploymentsOptions{
+			ListOptions: request.ListOptions{
+				AllPages: false,
+				Limit:    30,
+				Page:     0,
+			},
+			ChannelUpdate:       false,
+			DanglingDeployments: false,
+		}
+
+		deployments, err := service.ListDeploymentsForCluster(c, deployListOpts, cluster.Id)
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintf(opts.f.IOStreams.Out, "Got %d deployments in cluster %s\n", deployments.Size, cluster.Id)
+
+		for _, deploy := range deployments.Items {
+			snapshots = append(snapshots, snapshot.DeploymentSnapshot{
+				ClusterId: cluster.Id,
+				Id:        deploy.Id,
+				Kind:      deploy.Spec.ConnectorTypeId,
+				Status:    deploy.Status.Phase,
+			})
+		}
+	}
+
+	buffer := &bytes.Buffer{}
+
+	fmt.Fprintf(opts.f.IOStreams.Out, "Creating snapshot with %d connector deployments\n", len(snapshots))
+	err = dumper.DumpTable(
+		dumper.TableConfig[snapshot.DeploymentSnapshot]{
+			Style: dumper.TableStyleCSV,
+			Wide:  false,
+			Columns: []dumper.Column[snapshot.DeploymentSnapshot]{
+				{
+					Name: "ClusterID",
+					Wide: false,
+					Getter: func(in *snapshot.DeploymentSnapshot) dumper.Row {
+						return dumper.Row{
+							Value: in.ClusterId,
+						}
+					},
+				},
+				{
+					Name: "ID",
+					Wide: false,
+					Getter: func(in *snapshot.DeploymentSnapshot) dumper.Row {
+						return dumper.Row{
+							Value: in.Id,
+						}
+					},
+				},
+				{
+					Name: "ConnectorTypeID",
+					Wide: false,
+					Getter: func(in *snapshot.DeploymentSnapshot) dumper.Row {
+						return dumper.Row{
+							Value: in.Kind,
+						}
+					},
+				},
+				{
+					Name: "Status",
+					Wide: false,
+					Getter: func(in *snapshot.DeploymentSnapshot) dumper.Row {
+						return dumper.Row{
+							Value: string(in.Status),
+						}
+					},
+				},
+			},
+		},
+		buffer,
+		snapshots,
+	)
+	if err != nil {
+		return err
+	}
+
+	return runComparison(opts.f.IOStreams.Out, opts.inputFile, buffer)
+}
+
+func runComparison(out io.Writer, inputFile string, newSnapshot io.Reader) error {
```

Comment:
This is only food for thoughts.

I wonder if reading the files into bytes and using [bytes.Compare](http://golang.org/pkg/bytes/#Compare) is a better solution. 

You could also read the files content, md5 it and compare the two hashes.

None of these proposals can show the diff of the files, but it is a trade of I guess

Reply:
I'm going to change that code considerably, but keeping the diff is a must. The reason is that we want to run that command before and after a deploy as to make sure it didn't break anything, but it does break, we need to know what it was.

Reply:
Then your probably want to keep it as it is right now

Reply:
@rinaldodev I converted this to a draft to reflect the fact that there will be some refactoring

Reply:
Thanks